core/Filechecks.php

```diff
@@ -40,7 +40,9 @@ public static function checkDirectoriesWritable($directoriesToCheck)
     {
         $resultCheck = array();
         foreach ($directoriesToCheck as $directoryToCheck) {
-            if (!preg_match('/^' . preg_quote(PIWIK_USER_PATH, '/') . '/', $directoryToCheck)) {
+	        if (!preg_match('/^' . preg_quote(PIWIK_USER_PATH, '/') . '/', $directoryToCheck)
```

Comment:
those 3 constant might point to different directories so we need to only add another directory if none of these are set